source/simulator/core.cc

```diff
@@ -1178,6 +1151,16 @@ namespace aspect
 #else
     sp.compress();
 
+    // Deleting other matrix blocks (no need to have entries for
+    // constrained/hanging nodes). Note that no other matrices than c=0 are
+    // fully created anyways.
+    for (unsigned int c=1; c<introspection.n_compositional_fields; ++c)
+      {
+        const unsigned int block_idx = introspection.block_indices.compositional_fields[c];
+        sp.block(block_idx, block_idx).reinit(sp.block(block_idx, block_idx).locally_owned_range_indices(),sp.block(block_idx, block_idx).locally_owned_domain_indices()); // TODO: clearing it would be nice but clear() also resets the size
+        sp.block(block_idx, block_idx).compress();
```

Comment:
It took me a bit to understand the code. I think what you're doing is not to *delete* the block, but you're initializing it to a block of the correct size that simply has not entries within it? If so, maybe update the comment above.

Extra points if you wrap line 1160 :-)